### PR TITLE
add the session token into the header after the sign

### PR DESCRIPTION
**Pull request by shy950521**

### CODE REVIEW

**Review by twincitiesguy — CHANGES_REQUESTED** [review]
I don't see any v4 signer in the target branch - maybe this is targeting the wrong branch for v4 auth - but you will also need to add the logic to the v4 signer

File: src/main/java/com/emc/object/s3/S3SignerV2.java [hunk 1]
```diff
@@ -63,14 +63,19 @@ public S3SignerV2(S3Config s3Config) {
...
+                S3Constants.PARAM_SECURITY_TOKEN));
         if (s3Config.isSignMetadataSearch()) {
             signedParameters.add(S3Constants.PARAM_QUERY);
             signedParameters.add(S3Constants.PARAM_SEARCH_METADATA);
         }
     }
 
     public void sign(String method, String resource, Map<String, String> parameters, Map<String, List<Object>> headers) {
+
+        if (s3Config.getSessionToken() != null) {
```
[the remark this answers is not in the document]
Comment (shy950521) [reply src/main/java/com/emc/object/s3/S3SignerV2.java:76]: As I mentioned in the email, boto3 added token into params['SecurityToken'] in V2 signing, not sure whether the header works here. Still working on deploying the testbed, I will update it when I finished the test.

**Comment by shy950521** [comment]
> I don't see any v4 signer in the target branch - maybe this is targeting the wrong branch for v4 auth - but you will also need to add the logic to the v4 signer

the PR shared the same target branch of the V4 PR, I will update the V4 part when the v4 PR is merged

File: src/main/java/com/emc/object/s3/S3SignerV2.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (shy950521) [reply src/main/java/com/emc/object/s3/S3SignerV2.java:76]: moved to the headers.

**Comment by twincitiesguy** [comment]
what's the status of this draft?  has this already been merged by other PRs?

**Pull request closed without merging**